content/articles/2018-10-devtools-2.0.0.Rmarkdown

```diff
@@ -0,0 +1,228 @@
+devtools installs them all automatically. You will need to care, however, if
+you're filing a bug because reporting it at the correct place will lead to a
+speedier resolution.
+
+Package developers who wish to depend on devtools features should also pay
+attention to which package the functionality is coming from and depend on that
+rather than devtools. In most cases packages should not depend on devtools
+directly.
+
+## New features
+
+The majority of the work for this release was related to bugfixes and
+infrastructure improvements, but there are also some new features you may
+notice.
+
+As always a complete list of all the changes is available in the package
+[Changelog](http://devtools.r-lib.org/news/index.html).
+
+### Upgrade menu
+
+All of the install functions (e.g. `install_github()`) now prompt the user with
+a menu if there are dependencies of the package being installed which are
+outdated. This allows the user to pick which if any they would like to upgrade.
+Previous versions of devtools always upgraded these packages automatically by
+default, which sometimes was frustrating when you simply wanted to install one
+package.
+
+<p align="center">
+<img src="/images/devtools-2.0.0/upgrade-menu.png" alt="upgrade menu">
+</p>
+
+The menu respects the `menu.graphics` option, so set `options(menu.graphics =
+FALSE)` in your .Rprofile if you prefer text based menus, or `TRUE` if you
+prefer graphical widgets.
+
+### Improved check output
+
+`check()` now uses the [rcmdcheck] package, which has much richer (and
+colorful) output to the check results, which makes finding where a check fail
+much easier.
+
+<p align="center">
+<a href="https://github.com/r-lib/rcmdcheck">
+<img src="https://raw.githubusercontent.com/r-lib/rcmdcheck/master/inst/rcmdcheck.gif" alt="rcmdcheck output" width="700">
+</a>
+</p>
+
+### Testing single files
+
+devtools now includes functions (`test_file()` and `test_coverage_file()`) to
+improve development of a single file. Rather than running all
+tests, or manually supplying a `filter` argument to restrict the tests
+`test_file()` automatically runs the corresponding tests for a given source
+file. These functions make the feedback loop when developing new features
+quicker as you only run the relevant tests for the file you are editing.
+
+This requires you use a standard naming convention for your tests, e.g. if
+you have a source file `R/featureA.R` the corresponding test file would be
+`tests/testthat/test-featureR.R`.
+
+The tests file to run is automatically detected from the open file in RStudio (if
+available), so you can call `test_file()` with either the source file or
+the test file open. There is also a corresponding `test_coverage_file()` function, to show the test
+code coverage for a single source file, which works in the same way as
+`test_file()`.
+
+There are also RStudio addins for these functions so you can bind them to shortcut keys.
+We recommend
+
+- TODO for `test_file()`
```

Comment:
I don't have any good suggestions. I just type the function names every time 😢

Reply:
Ok, I will come up with some then!